packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/webview_controller_delegate.dart

```diff
@@ -0,0 +1,264 @@
+  /// [WebViewControllerDelegate].
+  ///
+  /// Should only be used by platform implementations because they can't extend
+  /// a class that only contains a factory constructor.
+  @protected
+  WebViewControllerDelegate.implementation() : super(token: _token);
+
+  static final Object _token = Object();
+
+  /// Loads the file located on the specified [absoluteFilePath].
+  ///
+  /// The [absoluteFilePath] parameter should contain the absolute path to the
+  /// file as it is stored on the device. For example:
+  /// `/Users/username/Documents/www/index.html`.
+  ///
+  /// Throws an ArgumentError if the [absoluteFilePath] does not exist.
+  Future<void> loadFile(
+    String absoluteFilePath,
+  ) {
+    throw UnimplementedError(
+        'WebView loadFile is not implemented on the current platform');
+  }
+
+  /// Loads the Flutter asset specified in the pubspec.yaml file.
+  ///
+  /// Throws an ArgumentError if [key] is not part of the specified assets
+  /// in the pubspec.yaml file.
+  Future<void> loadFlutterAsset(
+    String key,
+  ) {
+    throw UnimplementedError(
+        'WebView loadFlutterAsset is not implemented on the current platform');
+  }
+
+  /// Loads the supplied HTML string.
+  ///
+  /// The [baseUrl] parameter is used when resolving relative URLs within the
+  /// HTML string.
+  Future<void> loadHtmlString(
+    String html, {
+    String? baseUrl,
+  }) {
+    throw UnimplementedError(
+        'WebView loadHtmlString is not implemented on the current platform');
+  }
+
+  /// Loads the specified URL.
+  ///
+  /// If `headers` is not null and the URL is an HTTP URL, the key value paris in `headers` will
+  /// be added as key value pairs of HTTP headers for the request.
+  ///
+  /// `url` must not be null.
+  ///
+  /// Throws an ArgumentError if `url` is not a valid URL string.
+  Future<void> loadUrl(
+    String url,
+    Map<String, String>? headers,
+  ) {
+    throw UnimplementedError(
+        'WebView loadUrl is not implemented on the current platform');
+  }
+
+  /// Makes a specific HTTP request ands loads the response in the webview.
+  ///
+  /// [WebViewRequest.method] must be one of the supported HTTP methods
+  /// in [WebViewRequestMethod].
+  ///
+  /// If [WebViewRequest.headers] is not empty, its key-value pairs will be
+  /// added as the headers for the request.
+  ///
+  /// If [WebViewRequest.body] is not null, it will be added as the body
+  /// for the request.
+  ///
+  /// Throws an ArgumentError if [WebViewRequest.uri] has empty scheme.
+  Future<void> loadRequest(
+    LoadRequestParamsDelegate params,
+  ) {
+    throw UnimplementedError(
+        'WebView loadRequest is not implemented on the current platform');
+  }
+
+  /// Updates the webview settings.
+  ///
+  /// Any non null field in `settings` will be set as the new setting value.
+  /// All null fields in `settings` are ignored.
+  Future<void> updateSettings(WebSettingsDelegate setting) {
```

Comment:
This method can be removed. We want to create a new separate method for each `WebSetting` field. And then the `WebSettings` class can be removed too.

Reply:
Just to be sure, does this mean we should create separate getters and setters for each of the `WebSetting` fields in the `WebViewControllerDelegate` class?

Reply:
We should only need the setters for now. We can add getters at a later point if people want support for them.

Reply:
Also, the fields `hasNavigationDelegate` and `hasProgressTracking` can be ignored. A platform will be informed to this when `NavigationCallbackhandler.setOnNavigationRequest` is called by the app-facing interface.

Reply:
Fixed with commit [dca96f0](https://github.com/flutter/plugins/pull/5109/commits/dca96f07b2f95add924e2ec611be925017807b76)